faucet.go

```diff
@@ -948,21 +895,41 @@ func (l *lightningFaucet) payInvoice(homeTemplate *template.Template,
 	homeState *homePageContext, w http.ResponseWriter, r *http.Request) {
 
 	// Disable pay invoice if user set this parameter
-	if l.disablePayInvoices {
+	if l.cfg.DisablePayInvoices {
 		http.Error(w, "invoices payment was disabled", 403)
 		return
 	}
 
-	elapsed := time.Since(lastPayInvoiceTime)
-	if elapsed < PayInvoiceTimeout {
-		homeState.SubmissionError = PayInvoiceTimeNotElapsed
+	// Verify IP before open a channel
+	hostIP, err := getRealIP(r, l.cfg.UseRealIP)
+	if err != nil {
+		log.Errorf("Can't get client ip: %v", err)
+		homeState.SubmissionError = InternalServerError
 		homeTemplate.Execute(w, homeState)
 		return
 	}
-	lastPayInvoiceTime = time.Now()
 
-	// Get the invoice and try to verify and decode.
+	// Get the invoice from users form and set the input value again
+	// for the users repeat the action or verify
 	rawPayReq := r.FormValue("payinvoice")
+	homeState.FormFields["Payinvoice"] = rawPayReq
+
+	if err = verifyTimeLimit(hostIP, l.cfg.ActionsTimeLimit, l.rateLimitMtx); err != nil {
+		log.Errorf(fmt.Sprintf("%v", err))
+		homeState.SubmissionError = TimeLimitError
+		homeTemplate.Execute(w, homeState)
+		return
+	}
+
+	// Disable pay invoice if user set this parameter
+	if l.cfg.DisablePayInvoices {
+		http.Error(w, "invoices payment was disabled", 403)
+		return
+	}
+
+	lastPayInvoiceTime = time.Now()
```

Comment:
We dont need this global var any more because we record it per client in `requestIPs`